### PR TITLE
Clean up screen output

### LINKED ISSUE #23: Factoring output
Factoring demo output is incredibly long. Perhaps do some clean up?
```
Enter a number to be factored:
Input product        ( 0 <= P <= 63): 24
Running on QPU
{'numberOfReads': 50,
 'results': [{'a': 7,
              'b': 4,
              'numOfOccurrences': 9,
              'percentageOfOccurrences': 18.0,
              'valid': False},
             {'a': 5,
              'b': 4,
              'numOfOccurrences': 2,
              'percentageOfOccurrences': 4.0,
              'valid': False},
             {'a': 4,
              'b': 5,
              'numOfOccurrences': 1,
              'percentageOfOccurrences': 2.0,
              'valid': False},
             {'a': 6,
              'b': 7,
              'numOfOccurrences': 6,
              'percentageOfOccurrences': 12.0,
              'valid': False},
             {'a': 7,
              'b': 6,
              'numOfOccurrences': 7,
              'percentageOfOccurrences': 14.0,
              'valid': False},
             {'a': 3,
              'b': 6,
              'numOfOccurrences': 1,
              'percentageOfOccurrences': 2.0,
              'valid': False},
             {'a': 6,
              'b': 3,
              'numOfOccurrences': 1,
              'percentageOfOccurrences': 2.0,
              'valid': False},
             {'a': 5,
              'b': 6,
              'numOfOccurrences': 4,
              'percentageOfOccurrences': 8.0,
              'valid': False},
             {'a': 6,
              'b': 5,
              'numOfOccurrences': 7,
              'percentageOfOccurrences': 14.0,
              'valid': False},
             {'a': 4,
              'b': 7,
              'numOfOccurrences': 3,
              'percentageOfOccurrences': 6.0,
              'valid': False},
             {'a': 3,
              'b': 4,
              'numOfOccurrences': 1,
              'percentageOfOccurrences': 2.0,
              'valid': False},
             {'a': 6,
              'b': 1,
              'numOfOccurrences': 1,
              'percentageOfOccurrences': 2.0,
              'valid': False},
             {'a': 2,
              'b': 5,
              'numOfOccurrences': 1,
              'percentageOfOccurrences': 2.0,
              'valid': False},
             {'a': 7,
              'b': 3,
              'numOfOccurrences': 1,
              'percentageOfOccurrences': 2.0,
              'valid': False},
             {'a': 7,
              'b': 2,
              'numOfOccurrences': 2,
              'percentageOfOccurrences': 4.0,
              'valid': False},
             {'a': 4,
              'b': 0,
              'numOfOccurrences': 1,
              'percentageOfOccurrences': 2.0,
              'valid': False},
             {'a': 4,
              'b': 6,
              'numOfOccurrences': 1,
              'percentageOfOccurrences': 2.0,
              'valid': True},
             {'a': 6,
              'b': 6,
              'numOfOccurrences': 1,
              'percentageOfOccurrences': 2.0,
              'valid': False}],
 'timing': {'actual': {'qpuProcessTime': 25160}}}

```

### PR DESCRIPTION
**Pull request by mcfarljm**
Instead of printing the full output dictionary, which spans many
lines, print a formatted table showing the identified factors, whether
they are valid, the percentage occurrence and a pseudo bar-chart
representing occurrence.

Closes #23.

Previous output was extremely verbose and difficult to parse visually.  New output looks like:
```
Enter a number to be factored:
Input product        ( 0 <= P <= 63): 24
Running on QPU
--------------------------------------------
Factors    Valid?  Percentage of occurrences
--------------------------------------------
(  6,  4)  Yes      12 ######
(  4,  6)  Yes       2 #
(  7,  4)            5 ##
(  4,  7)            1 
(  5,  4)            1 
(  4,  5)            5 ##
(  4,  4)            5 ##
(  6,  7)            3 #
(  3,  6)            3 #
(  6,  3)            3 #
(  6,  6)           12 ######
(  6,  5)            5 ##
(  2,  7)            7 ###
(  3,  4)            7 ###
(  6,  2)            2 #
(  2,  6)            7 ###
(  4,  3)            2 #
(  2,  3)            2 #
(  2,  4)            2 #
(  1,  4)            1 
(  0,  3)            1 
(  7,  2)            2 #
(  2,  5)            4 ##
(  2,  2)            1 
(  4,  0)            1 
(  0,  0)            1 
(  5,  6)            1 
(  0,  7)            1 
(  6,  0)            1 
```

### CODE REVIEW

**Review by arcondello — APPROVED** [review]

File: demo.py [hunk 1]
```diff
@@ -146,6 +145,22 @@ def factor(P):
 
     return output
 
+def display_output(output):
+    header_str = 'Factors    Valid?  Percentage of occurences'
```
Comment (randomir) [inline demo.py]: Typo: occurences.

Comment (JoelPasvolsky) [inline demo.py]: Occurrences 
(double r, capitalized O)

**Review by randomir — APPROVED** [review]
New output looks great!

**Review by JoelPasvolsky — APPROVED** [review]
This is a very nice improvement. 
One minor suggestions: even having read your comment "a pseudo bar-chart" when I looked at the printed output, I first thought the hashes were a formatting problem on percentage signs. 
Perhaps (1) change to a different char like | or - or * and (2) add a 2nd header line like this:

---
Factors    Valid?  Percentage of Occurrences
  shifted right  --->                        [Numerical & Graphic Representation]

---

**Comment by mcfarljm** [comment]
Thanks for the reviews.  New output looks like:
```
---------------------------------------------------
Factors    Valid?  Percentage of Occurrences
                   Numeric & Graphic Representation
---------------------------------------------------
(  6,  4)  Yes       5 **
(  4,  6)  Yes      14 *******
(  5,  4)            1 
(  4,  2)            9 *****
(  7,  6)            2 *
```

**Comment by randomir** [comment]
Since we're bikeshedding now, I like the progress bar style common on linux command line:
```
(  6,  4)  Yes      12 [######.....]
(  4,  6)  Yes       2 [#..........]
(  7,  4)            5 [##.........]
(  4,  7)            1 [...........]
(  5,  4)            1 [...........]
(  4,  5)            5 [##.........]
```

**Comment by mcfarljm** [comment]
> 
> 
> Since we're bikeshedding now, I like the progress bar style common on linux command line:
> 
> ```
> (  6,  4)  Yes      12 [######.....]
> (  4,  6)  Yes       2 [#..........]
> (  7,  4)            5 [##.........]
> (  4,  7)            1 [...........]
> (  5,  4)            1 [...........]
> (  4,  5)            5 [##.........]
> ```

Here is a draft of what that would look like if I keep the same total allowed width:
```
---------------------------------------------------
Factors    Valid?  Percentage of Occurrences
                   Numeric & Graphic Representation
---------------------------------------------------
(  6,  4)  Yes       8 [####.....................................................]
(  4,  6)  Yes      14 [#######..................................................]
(  7,  4)            2 [#........................................................]
(  5,  4)            2 [#........................................................]
(  4,  4)            3 [#........................................................]
```
What do you think?  The quirk is that the occurrence percentages tend to be somewhat smallish (< 15% or so), but I'm also trying to leave space for the possibility that they could be as large as 100%.

**Comment by JoelPasvolsky** [comment]
What if the granularity is 5%?

**Comment by randomir** [comment]
Let's just keep it simple. The stars were perfectly good enough.

**Comment by JoelPasvolsky** [comment]
Partypooper

**Comment by mcfarljm** [comment]
FWIW, example with 5% increments:
```
---------------------------------------------------
Factors    Valid?  Percentage of Occurrences
                   Numeric & Graphic Representation
---------------------------------------------------
(  6,  4)  Yes       6 [#...................]
(  4,  6)  Yes       2 [....................]
(  4,  7)            2 [....................]
(  5,  4)            5 [#...................]
(  4,  5)            1 [....................]
(  4,  4)            1 [....................]
```

**Comment by mcfarljm** [comment]
If there are no objections, I suggest we go ahead and merge as is

**Pull request merged**